src/parameterizations/lateral/MOM_mixed_layer_restrat.F90

```diff
@@ -970,6 +984,26 @@ subroutine mixedlayer_restrat_Bodner(CS, G, GV, US, h, uhtr, vhtr, tv, forces, d
     CS%wpup_filtered(i,j) = wpup(i,j)
   enddo ; enddo
 
+  if (CS%id_lfbod > 0) then
+    do j=js-1,je+1 ; do i=is-1,ie+1
+      ! Calculate front length used in B22 formula (eq 24).
+      w_star3 = max(0., -bflux(i,j)) * BLD(i,j)
+      u_star3 = U_star_2d(i,j)**3
+
+      ! Include an absurdly_small_freq2 to prevent division by zero.
+      f_h = 0.25 * ((G%CoriolisBu(I,J)  + G%CoriolisBu(I-1,J-1)) &
+          + (G%CoriolisBu(I-1,J) + G%CoriolisBu(I,J-1)))
+      f2_h = max(f_h**2, absurdly_small_freq2)
+
+      lf_bodner_diag(i,j) = &
+          0.25 * cuberoot(CS%mstar * u_star3 + CS%nstar * w_star3)**2 &
+            / (f2_h * max(little_h(i,j), GV%Angstrom_H))
+
+      ! Rescale from [Z2 H-1 to L]
+      lf_bodner_diag(i,j) = lf_bodner_diag(i,j) * uflux_rescale
+    enddo ; enddo
```

Comment:
This line has a horizontal indexing error in non-Boussinesq mode because in that case the appropriate rescaling factor is a function of `(i,j)`.